action.yml

```diff
@@ -302,6 +310,50 @@ runs:
           echo "Running PHPStan"
           vendor/bin/phpstan analyse
         fi
+        # Validate licenses of all composer dependencies are allowed
+        echo "Checking licenses of all dependencies"
+        composer global require madewithlove/license-checker
+        COMPOSER_GLOBAL_HOME=$(composer -q -n config --global home)
+        SPDX_ALLOWED_DELIMITED=$(cat ${{ github.action_path }}/allowed-spdx-delimited.txt | tr -d '\n')
+        # Update the licenses in the installed.json file to be sorted so that allowed SPDX identifier
+        # are at the top of the list. This is done because the license-checker will only check the first SPDX.
+        SPDX_ALLOWED_DELIMITED=$SPDX_ALLOWED_DELIMITED php -r '
+          $allowedSpdxDelimted = getenv("SPDX_ALLOWED_DELIMITED");
+          $allowedSpdx = explode(";", $allowedSpdxDelimted);
+          $filename = "vendor/composer/installed.json";
+          $contents = file_get_contents("vendor/composer/installed.json");
+          $json = json_decode($contents, true);
+          foreach ($json["packages"] as &$package) {
+            # A handful of silverstripe packages do not have a license field, though we do not need to check them
+            if (str_starts_with($package["name"], "silverstripe/")) {
+              continue;
+            }
```

Comment:
We should add the license field wherever it's missing.

Reply:
Have created PRs to do this https://github.com/silverstripe/silverstripe-installer/pull/382 https://github.com/silverstripe/supported-modules/pull/50